src/TiciTacaToeyGameEngine.ts

```diff
@@ -408,6 +413,21 @@ class TiciTacaToeyGameEngine implements GameEngine {
   }
 }
 
+const addPlayer = (
+  players: {
+    [key: string]: ConnectedPlayer;
+  },
+  playerId: string,
+  name: string,
+  connection: WebSocket
+  ): any => {
+  players = {
```

Comment:
To avoid mutation - lets just return this object instead of assigning it to the input variable

return {
    ...players,
    [playerId]: { "playerId": playerId, "name": name, "connection": connection },
  };

Reply:
makes sense!